dev/release.sh

```diff
@@ -0,0 +1,614 @@
+#! /bin/sh -e
+# Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+usage () {
+    cat <<EOF
+Usage: release.sh [ options ... ]
+
+--alpha         Start or increase the "alpha" pre-release tag.
+--next-beta     Switch to the "beta" pre-release tag after alpha release.
+                It can only be given with --alpha.
+--beta          Start or increase the "beta" pre-release tag.
+--final         Get out of "alpha" or "beta" and make a final release.
+                Implies --branch.
+
+--branch        Create a release branch 'openssl-{major}.{minor}.x',
+                where '{major}' and '{minor}' are the major and minor version
+                numbers.
+
+--no-upload     Don't upload to upload@dev.openssl.org.
+--no-update     Don't perform 'make update'.
+--verbose       Verbose output.
+--debug         Include debug output.  Implies --no-upload.
+
+--force         Force execution
+
+--help          This text
+--manual        The manual
+
+If none of --major, --minor, --alpha, --beta, or --final are given, this script
+tries to figure out the next step.
+EOF
+    exit 0
+}
+
+# Set to one of 'major', 'minor', 'alpha', 'beta' or 'final'
+next_method=
+next_method2=
+
+do_branch=false
+
+do_clean=true
+do_upload=true
+do_update=true
+DEBUG=:
+VERBOSE=:
+git_quiet=-q
+
+force=false
+
+do_help=false
+do_manual=false
+
+TEMP=$(getopt -l 'alpha,next-beta,beta,final' \
+              -l 'branch' \
+              -l 'no-upload,no-update' \
+              -l 'verbose,debug' \
+              -l 'force' \
+              -l 'help,manual' \
+              -n release.sh -- - "$@")
+eval set -- "$TEMP"
+while true; do
+    case $1 in
+        --alpha | --beta | --final )
+            next_method=$(echo "x$1" | sed -e 's|^x--||')
+            if [ -z "$next_method2" ]; then
+                next_method2=$next_method
+            fi
+            shift
+            if [ "$next_method" = 'final' ]; then
+                do_branch=true
```

Comment:
Lines around 206.  A warning, but otherwise inconsequential.  So what I hear is that maybe I should use a different variable name for the implied flags...

Reply:
Done, I simply added another variable, `warn_branch`, to tell if the warning should be displayed or not.